src/platform/Darwin/DnssdImpl.cpp

```diff
@@ -299,8 +299,9 @@ static CHIP_ERROR Resolve(void * context, DnssdResolveCallback callback, uint32_
     return Resolve(sdCtx, interfaceId, addressType, type, name);
 }
 
-static CHIP_ERROR Resolve(CommissioningResolveDelegate * delegate, uint32_t interfaceId, chip::Inet::IPAddressType addressType,
-                          const char * type, const char * name)
+template <typename T>
+static CHIP_ERROR Resolve(T delegate, uint32_t interfaceId, chip::Inet::IPAddressType addressType, const char * type,
```

Comment:
```suggestion
static CHIP_ERROR Resolve(T * delegate, uint32_t interfaceId, chip::Inet::IPAddressType addressType, const char * type,
```
To make it clearer that it's expected to be a pointer.